spyder/app/mainwindow.py

```diff
@@ -3431,7 +3431,8 @@ def show_tour_message(self):
         """
         Show message about starting the tour the first time Spyder starts.
         """
-        if not running_under_pytest():
+        dev_mode = os.environ.get('SPYDER_DEV')
+        if not running_under_pytest() and not dev_mode:

```

Comment:
```suggestion
        if not running_under_pytest() and not DEV:
```

The line you introduced is not necessary. As I told you in my last comment, we declare a global variable that does exactly that in `spyder/config/base.py`:

https://github.com/spyder-ide/spyder/blob/9c422b81f7b839b79f88d50b0fc60e849be8cc4c/spyder/config/base.py#L39

And `DEV` is imported at the top of this file:

https://github.com/spyder-ide/spyder/blob/3ea0bdbbef55de203aabe800384e4fa811fb7434/spyder/app/mainwindow.py#L148

So we can use it here directly.

Reply:
Sorry, I was in the process of writing this when the other suggestions got applied.

Actually, this will prevent us from actually testing it, manually or otherwise. I was in the process of submitting a revised suggestion that instead checks for the safe mode env variable, so that if we're intentionally running Spyder under safe mode then it will not nag us (or the user) with the spurious popup, while simply running Spyder under the normal dev environment with settings reset will show it as expected, which we need to do anyway to properly test the only-run-once behavior.

```suggestion
        if (not running_under_pytest()
                and not os.environ.get('SPYDER_SAFE_MODE')):
```

Reply:
lol sorry. Fixing that now

Reply:
> Actually, this will make it impossible for us to actually test the popup, manually or automatically

Ok, fair enough. However, the same I said about `SPYDER_DEV` applies for `SPYDER_SAFE_MODE`: we already declare a global variable for that in `base/config.py`.